src/decision/test/lidar-decision-test.cpp

```diff
@@ -6,10 +6,110 @@
 
 #include <LidarDecision.h>
 #include <gtest/gtest.h>
+#include <vector>
+
+TEST(LidarDecision, obstacle_in_range){
+    sensor_msgs::LaserScan test_scan;
+    test_scan.angle_max = 3.1415;
+    test_scan.angle_min = 0.0000;
+    test_scan.angle_increment = 0.01;
+    std::vector<float> fake_test_data(300);
+
+    for (int i = 0;i < fake_test_data.size();i++){
+        fake_test_data[i] = 999999;
+    }
+    for (int i=100;i<150;i++)
+        fake_test_data[i]=13;
+    test_scan.ranges = fake_test_data;
+    sensor_msgs::LaserScan::ConstPtr test_scan_ptr(new sensor_msgs::LaserScan(test_scan));
+
+    EXPECT_EQ(false, LidarDecision::obstacle_in_range(1.00, 1.50, 5, test_scan_ptr));
+    //obstacle inside of angle range but outside distance
+    EXPECT_EQ(true, LidarDecision::obstacle_in_range(1.00, 1.50, 15, test_scan_ptr));
+    //in angle range within distance
+    EXPECT_EQ(false, LidarDecision::obstacle_in_range(1.51, 2.50, 15, test_scan_ptr));
+    //within distance but just outside of angle range
+    EXPECT_EQ(false, LidarDecision::obstacle_in_range(2.00, 2.55, 5, test_scan_ptr));
+    //out side distance and out of angle
+    EXPECT_EQ(true, LidarDecision::obstacle_in_range(1.48, 1.7, 15, test_scan_ptr));
+    // partly in range
+
+
+}
+
+TEST(LidarDecision, manage_twist){
```

Comment:
All 4 of the tests in here should be refactored into their own tests. You should also use gtest's setup functionality so you don't have a bunch of duplicate code at the start of each test (See [here](https://github.com/google/googletest/blob/master/googletest/docs/Primer.md#test-fixtures-using-the-same-data-configuration-for-multiple-tests) for a quick runthrough on how to do this)